streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -564,8 +574,16 @@ private void restoreChangelog(final ChangelogMetadata changelogMetadata) {
             return Collections.emptyMap();
 
         try {
-            return restoreConsumer.endOffsets(partitions);
-        } catch (final TimeoutException e) {
+            if (adminClient != null) {
+                final ListOffsetsResult result = adminClient.listOffsets(partitions.stream().collect(
+                        Collectors.toMap(Function.identity(), tp -> OffsetSpec.latest())));
+                return result.all().get().entrySet().stream().collect(
+                        Collectors.toMap(Map.Entry::getKey, entry -> entry.getValue().offset()));
+            } else {
+                // we only fall back to use restore consumer if admin client is not set in TTD
```

Comment:
Can this case ever apply? TTD should never restore any task from my understanding.

Reply:
TTD would still call this function though... after some thinking I've decided to add a mock-admin to streams:test-utils (the existing mock is in clients:test and hence cannot be relied on).